app/src/main/java/org/sopt/and/feature/signup/SignUpScreen.kt

```diff
@@ -0,0 +1,260 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.flowWithLifecycle
+import org.sopt.and.R
+import org.sopt.and.core.designsystem.component.SocialLoginButton
+import org.sopt.and.core.designsystem.component.textfield.EmailTextField
+import org.sopt.and.core.designsystem.component.textfield.PasswordTextField
+import org.sopt.and.core.designsystem.theme.ANDANDROIDTheme
+import org.sopt.and.core.extension.toast
+
+@Composable
+fun SignUpRoute(
+    navigateToSignIn: (String, String) -> Unit,
+    viewModel: SignUpViewModel = hiltViewModel(),
+    modifier: Modifier = Modifier,
+) {
+    val signUpState by viewModel.signUpState.collectAsState()
+
+    val context = LocalContext.current
+    val lifecycleOwner = LocalLifecycleOwner.current
+
+    LaunchedEffect(viewModel.signUpSideEffect, lifecycleOwner) {
+        viewModel.signUpSideEffect.flowWithLifecycle(lifecycle = lifecycleOwner.lifecycle)
+            .collect { sideEffect ->
+                when (sideEffect) {
+                    is SignUpSideEffect.Toast -> context.toast(sideEffect.message)
+                    is SignUpSideEffect.NavigateToSignIn -> {
+                        context.toast(R.string.sign_up_success)
+                        navigateToSignIn(signUpState.email, signUpState.password)
+                    }
+                }
+            }
+    }
+
+    SignUpScreen(
+        onSignUpButtonClick = viewModel::onSignUpClick,
+        onIdChange = viewModel::updateEmail,
+        onPasswordChange = viewModel::updatePassword,
+        modifier = modifier,
+    )
+
+}
+
+@Composable
+fun SignUpScreen(
+    onSignUpButtonClick: (String, String) -> Unit,
+    onIdChange: (String) -> Unit,
+    onPasswordChange: (String) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val email = remember { mutableStateOf("") }
+    val password = remember { mutableStateOf("") }
```

Comment:
1)

```suggestion
    var email by remember { mutableStateOf("") }
    var password by remember { mutableStateOf("") }
```

이렇게 선언하시면 하단에서 email 변수를 사용해서 state를 변경하실 수 있어요.

2)

ViewModel에서도 상태를 관리하고 여기서도 email, pw 상태관리를 하시네요. 중복된 상태관리는 상태 불일치의 가능성을 높이기 때문에 최대한 상태관리는 한 곳에서 이뤄져야합니다. 

그에 대한 증빙으로

```kotlin
@Composable
fun SignUpScreen(
    onSignUpButtonClick: (String, String) -> Unit,
    onIdChange: (String) -> Unit,
    onPasswordChange: (String) -> Unit,
    modifier: Modifier = Modifier,
) {
```

지금 뷰모델에 있는 상태관리를 위해 이렇게 패러미터들을 받아오고 계신데 로컬 변수에서 `mutableStateOf`를 사용하신 것이라면 패러미터에 있는 `onIdChange`를 사용하지 말고 email의 프로퍼티를 변경해야하지 않을까요?

생각이 너무 많이 들어가있는 코드인 것 같습니다. 어떤 상태들을 주로 변경해야하는지 생각해보시면 좋을 것 같아요.

Reply:
우왕 조언 감사합니다..!! 말씀하신대로 상태관리를 viewModel과 Screen에서 동시에 하고 있었네요!! 다시 한번 현재 관리중인 상태들을 정리해보는 것이 필요할 것 같아요 감사합니다😊